tests/Shared/WorkloadTesting/RequiresDockerDiscoverer.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Components.Common.Tests;
+using Microsoft.DotNet.XUnitExtensions;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Aspire.Workload.Tests;
+
+public class RequiresDockerDiscoverer : ITraitDiscoverer
+{
+    public IEnumerable<KeyValuePair<string, string>> GetTraits(IAttributeInfo traitAttribute)
+    {
+        if (!RequiresDockerTheoryAttribute.IsSupported)
```

Comment:
(not necessarily for this PR) - it would be nice if we had a consolidated "RequiresDocker" story across all our tests. This feels pretty spread out - some in `Aspire.Components.Common.Tests`, some here, some in `Aspire.Hosting.Tests`.

Reply:
Tracking in https://github.com/dotnet/aspire/issues/4222 .